pkg/util/net/net.go

```diff
@@ -72,3 +89,163 @@ func setBuffers(rc syscall.RawConn, sz int) error {
 
 	return err
 }
+
+func RemovePrivateDNSZone(
+	ctx context.Context,
+	log *logrus.Entry,
+	privateZonesClient privatedns.PrivateZonesClient,
+	configcli configclient.Interface,
+	mcocli mcoclient.Interface,
+	kubernetescli kubernetes.Interface,
+	vNetLinksClient privatedns.VirtualNetworkLinksClient,
+	resourceGroupID string) error {
+	if privateZonesClient == nil {
```

Comment:
Yeah, checking an interface against nil is a pretty common thing. _Can we push back on the use of interfaces?_ -> Passing an interface vs passing a struct have different use cases. We usually pass an interface in the tests when we want to mock that interface because it is a dependency that some function or constructor requires and want to control what it returns. As you mention, in some cases changing an interface by a struct is not a win as with the struct you still need to compare against an empty struct of that type (sometimes people use a pointer to the struct to check if it is empty or not). 

In our case is a bit different because the struct that has the same name as the interface is just a wrapper of our Azure clients that exists for mocking purposes (if I am not mistaken) +  it contains some extra functionality on top of the standard mgmtprivatedns.PrivateZonesClient (pkg/util/azureclient/mgmt/privatedns/privatezones.go). It is a common pattern you will see in our clients.

Reply:
Thanks for the explanation. I can't say I love it but I guess I'll learn to live with it ;)